nexus/reconfigurator/planning/src/blueprint_builder/internal_dns.rs

```diff
@@ -103,62 +101,68 @@ pub mod test {
     use super::*;
     use crate::blueprint_builder::test::verify_blueprint;
     use crate::example::ExampleSystem;
-    use omicron_common::policy::{
-        INTERNAL_DNS_REDUNDANCY, MAX_INTERNAL_DNS_REDUNDANCY,
-    };
+    use nexus_types::deployment::BlueprintZoneFilter;
+    use omicron_common::policy::INTERNAL_DNS_REDUNDANCY;
     use omicron_test_utils::dev::test_setup_log;
 
     #[test]
     fn test_dns_subnet_allocator() {
         static TEST_NAME: &str = "test_dns_subnet_allocator";
         let logctx = test_setup_log(TEST_NAME);
 
+        // Our test below assumes `INTERNAL_DNS_REDUNDANCY` is greater than 1
+        // (so we can test adding more); fail fast if that changes.
+        assert!(INTERNAL_DNS_REDUNDANCY > 1);
+
         // Use our example system to create a blueprint and input.
-        let example =
+        let mut example =
             ExampleSystem::new(&logctx.log, TEST_NAME, INTERNAL_DNS_REDUNDANCY);
-        let blueprint1 = &example.blueprint;
+        let blueprint1 = &mut example.blueprint;
+
+        // `ExampleSystem` adds an internal DNS server to every sled. Manually
+        // prune out all but the first of them to give us space to add more.
+        for (_, zone_config) in blueprint1.blueprint_zones.iter_mut().skip(1) {
+            zone_config.zones.retain(|z| !z.zone_type.is_internal_dns());
+        }
+        let npruned = blueprint1.blueprint_zones.len() - 1;
+
         verify_blueprint(blueprint1);
 
         // Create an allocator.
-        let mut allocator = DnsSubnetAllocator::new(blueprint1, &example.input)
-            .expect("can't create allocator");
-
-        // Save the first & last allocated subnets.
+        let mut allocator = DnsSubnetAllocator::new(
+            blueprint1
+                .all_omicron_zones(BlueprintZoneFilter::ShouldBeRunning)
+                .map(|(_sled_id, zone_config)| zone_config),
+            &example.input,
+        )
+        .expect("can't create allocator");
+
+        // There should be exactly one subnet allocated.
+        assert_eq!(allocator.len(), 1, "should be 1 subnets allocated");
         let first = allocator.first().expect("should be a first subnet");
-        let last = allocator.last().expect("should be a last subnet");
-        assert!(last > first, "first should come before last");
-
-        // Derive the reserved rack subnet.
+        let mut last = allocator.last().expect("should be a last subnet");
+        assert_eq!(first, last);
         let rack_subnet = first.rack_subnet();
-        assert_eq!(
-            rack_subnet,
-            last.rack_subnet(),
-            "first & last DNS subnets should be in the same rack subnet"
-        );
 
-        // Allocate two new subnets.
-        assert_eq!(MAX_INTERNAL_DNS_REDUNDANCY - INTERNAL_DNS_REDUNDANCY, 2);
+        // Allocate `npruned` new subnets.
+        for _ in 0..npruned {
```

Comment:
Is it worth asserting that `npruned > 0` just to make sure the test actually did something?  I guess if it didn't, we'd probably fail either the assertion that `INTERNAL_DNS_REDUNDANCY > 1` or the one at L161.

Reply:
We would but that isn't obvious. Easy enough to add that assertion.